File: tests/bench/01_recursion.go
// Benchmark 01: Recursion — naive recursive Fibonacci
// Measures function call overhead, stack frame creation
package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	n := 45
	start := time.Now()
	r := fib(n)
	elapsed := time.Since(start).Milliseconds()
	fmt.Printf("[bench] recursion: fib(%d) = %d in %d ms\n", n, r, elapsed)
}

File: tests/bench/01_recursion.rs
// Benchmark 01: Recursion — naive recursive Fibonacci
// Measures function call overhead, stack frame creation

use std::time::Instant;

fn fib(n: i64) -> i64 {
    if n < 2 { n } else { fib(n - 1) + fib(n - 2) }
}

fn main() {
    let n = 45;
    let start = Instant::now();
    let r = fib(n);
    let elapsed = start.elapsed().as_millis();
    println!("[bench] recursion: fib({}) = {} in {} ms", n, r, elapsed);
}
